politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -0,0 +1,292 @@
+// provided backend files. If a ProposalMetadata is not found, nil is returned.
+func proposalMetadataDecode(files []backend.File) (*pi.ProposalMetadata, error) {
+	var propMD *pi.ProposalMetadata
+	for _, v := range files {
+		if v.Name == pi.FileNameProposalMetadata {
+			b, err := base64.StdEncoding.DecodeString(v.Payload)
+			if err != nil {
+				return nil, err
+			}
+			var m pi.ProposalMetadata
+			err = json.Unmarshal(b, &m)
+			if err != nil {
+				return nil, err
+			}
+			propMD = &m
+			break
+		}
+	}
+	return propMD, nil
+}
+
+// proposalNameIsValid returns whether the provided name is a valid proposal
+// name.
+func (p *piPlugin) proposalNameIsValid(name string) bool {
+	return p.proposalNameRegexp.MatchString(name)
+}
+
+// proposalFilesVerify verifies the files adhere to all plugin setting
+// requirements. If this hook is being executed then the files have already
+// passed politeia validation so we can assume that the file has a unique name,
+// a valid base64 payload, and that the file digest and MIME type are correct.
+func (p *piPlugin) proposalFilesVerify(files []backend.File) error {
+	var imagesCount uint32
+	for _, v := range files {
+		payload, err := base64.StdEncoding.DecodeString(v.Payload)
+		if err != nil {
+			return fmt.Errorf("invalid base64 %v", v.Name)
+		}
+
+		// MIME type specific validation
+		switch v.MIME {
+		case mimeTypeText, mimeTypeTextUTF8:
+			// Verify text file is allowed
+			_, ok := allowedTextFiles[v.Name]
+			if !ok {
+				return backend.PluginError{
+					PluginID:     pi.PluginID,
+					ErrorCode:    uint32(pi.ErrorCodeTextFileNameInvalid),
+					ErrorContext: v.Name,
+				}
+			}
+
+			// Verify text file size
+			if len(payload) > int(p.textFileSizeMax) {
+				e := fmt.Sprintf("file %v size %v exceeds max size %v",
+					v.Name, len(payload), p.textFileSizeMax)
+				return backend.PluginError{
+					PluginID:     pi.PluginID,
+					ErrorCode:    uint32(pi.ErrorCodeTextFileSizeInvalid),
+					ErrorContext: e,
+				}
+			}
+
+		case mimeTypePNG:
+			imagesCount++
+
+			// Verify image file size
+			if len(payload) > int(p.imageFileSizeMax) {
+				e := fmt.Sprintf("image %v size %v exceeds max size %v",
+					v.Name, len(payload), p.imageFileSizeMax)
+				return backend.PluginError{
+					PluginID:     pi.PluginID,
+					ErrorCode:    uint32(pi.ErrorCodeImageFileSizeInvalid),
+					ErrorContext: e,
+				}
+			}
+
+		default:
+			return fmt.Errorf("invalid mime: %v", v.MIME)
+		}
+	}
+
+	// Verify that an index file is present
+	var found bool
+	for _, v := range files {
+		if v.Name == pi.FileNameIndexFile {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return backend.PluginError{
+			PluginID:     pi.PluginID,
+			ErrorCode:    uint32(pi.ErrorCodeTextFileMissing),
+			ErrorContext: pi.FileNameIndexFile,
+		}
+	}
+
+	// Verify image file count is acceptable
+	if imagesCount > p.imageFileCountMax {
+		e := fmt.Sprintf("got %v image files, max is %v",
```

Comment:
I'd skip creating the var and direct assign this in the struct.

Reply:
All of these have been fixed throughout politeiad.